tests/integration_tests/validation/discover_tests.py

```diff
@@ -1,30 +1,131 @@
+    'esef_xhtml_2022',
+    'esef_xhtml_2023',
+    'xbrl_calculations_1_1',
+    'xbrl_dimensions_1_0',
+    'xbrl_extensible_enumerations_1_0',
+    'xbrl_extensible_enumerations_2_0',
+    'xbrl_formula_1_0_assertion_severity_2_0',
+    'xbrl_formula_1_0_function_registry',
+    'xbrl_link_role_registry_1_0',
+    'xbrl_oim_1_0',
+    'xbrl_taxonomy_packages_1_0',
+    'xbrl_transformation_registry_3',
+    'xbrl_utr_malformed_1_0',
+    'xbrl_utr_registry_1_0',
+    'xbrl_utr_structure_1_0',
+}
 
 
 class Entry(TypedDict, total=False):
     name: str
     cache: bool
+    os: str
+    python_version: str
     shard: str
 
 
-output: list[Entry] = []
-for config in ALL_CONFORMANCE_SUITE_CONFIGS:
+def generate_config_entry(name: str, network_or_cache_required: bool, os: str, python_version: str, shard: str | None) -> Entry:
+    e: Entry = {
+        'name': name,
+        'cache': network_or_cache_required,
+        'os': os,
+        'python_version': python_version,
+    }
+    if shard is not None:
+        e['shard'] = shard
+    return e
+
+
+def generate_config_entries(config: ConformanceSuiteConfig, os: str, python_version: str, minimal: bool = False) -> Iterable[Entry]:
     if config.shards == 1:
-        output.append({
-            'name': config.name,
-            'cache': config.network_or_cache_required,
-        })
+        yield generate_config_entry(
+            name=config.name,
+            network_or_cache_required=config.network_or_cache_required,
+            os=os,
+            python_version=python_version,
+            shard=None,
+        )
     else:
-        for i in range(config.shards):
-            output.append({
-                'name': config.name,
-                'cache': config.network_or_cache_required,
-                'shard': str(i),
-            })
-json.dump(output, sys.stdout, indent=4)
-print()
+        ncores = OS_CORES[os]
+        shard_range = [0] if minimal else range(0, config.shards, ncores)
+        for start in shard_range:
+            end = min(config.shards, start + ncores)
+            yield generate_config_entry(
+                name=config.name,
+                network_or_cache_required=config.network_or_cache_required,
+                os=os,
+                python_version=python_version,
+                shard=f'{start}-{end}',
+            )
+
+
+def main() -> None:
+    output: list[Entry] = []
+    config_names_seen: set[str] = set()
+    for config in ALL_CONFORMANCE_SUITE_CONFIGS:
+        if config.name in FAST_CONFIG_NAMES:
+            assert not config.network_or_cache_required
+            assert config.shards == 1
+            config_names_seen.add(config.name)
+    assert not (FAST_CONFIG_NAMES - config_names_seen), \
+        f'Missing some fast configurations: {sorted(FAST_CONFIG_NAMES - config_names_seen)}'
+    for os in [LINUX, MACOS, WINDOWS]:
+        output.append(generate_config_entry(
+            name=','.join(sorted(FAST_CONFIG_NAMES)),
```

Comment:
I'm wondering if it's better to have a separate property for a shorthand label for display purposes. For example, the OS and Python version aren't visible even after expanding the logs:
![image](https://github.com/Arelle/Arelle/assets/105066394/005e2284-536f-40f6-a014-b248d81ea783)